File: src/dataset/cpp/bob_exercism-tqa236/claude/ClaudeSonnet4_bob_v1.cpp
#include "bob.h"

std::string bob::hey(std::string text) {
  bool no_lowercase = true;
  bool has_uppercase = false;
  bool is_question = false;
  bool is_empty = true;
  
  for (const char c : text) {
    if (!std::isspace(c)) {
      is_empty = false;
      if (c == '?') {
        is_question = true;
      } else {
        is_question = false;
      }
    }
    if (std::islower(c)) {
      no_lowercase = false;
    } else if (std::isupper(c)) {
      has_uppercase = true;
    }
  }
  
  if (is_empty) {
    return "Fine. Be that way!";
  }
  
  const bool is_yelling = no_lowercase && has_uppercase;
  if (is_yelling && is_question) {
    return "Calm down, I know what I'm doing!";
  }
  if (is_yelling) {
    return "Whoa, chill out!";
  }
  if (is_question) {
    return "Sure.";
  }
  return "Whatever.";
}

File: src/dataset/Java/bob_exercism-uzilan/claude/ClaudeSonnet4_Bob_v1.java
public class Bob {

    String hey(String s) {
        final String trimmed = s.trim();

        if (trimmed.isEmpty()) {
            return "Fine. Be that way!";
        }

        final boolean isQuestion = trimmed.charAt(trimmed.length() - 1) == '?';
        final boolean isUpperCase = isUpperCase(trimmed);

        if (isUpperCase) {
            if (isQuestion) {
                return "Calm down, I know what I'm doing!";
            } else {
                return "Whoa, chill out!";
            }
        }
        if (isQuestion) {
            return "Sure.";
        }
        return "Whatever.";
    }

    private boolean isQuestion(String s) {
        return s.charAt(s.length() - 1) == '?';
    }

    private boolean isUpperCase(String s) {
        boolean hasLetter = false;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (Character.isLetter(c)) {
                hasLetter = true;
                if (Character.isLowerCase(c)) {
                    return false;
                }
            }
        }
        return hasLetter;
    }
}